SimpleCalculator/SimpleCalculatorTest/Test.cs

```diff
@@ -0,0 +1,144 @@
+using SimpleCalculator.Business;
+using SimpleCalculator.Business.OperatorBusiness;
+using SimpleCalculator.Business.OperatorBusiness.Operators;
+using SimpleCalculator.Business.Abstraction;
+using SimpleCalculator.Business.Enums;
+using NSubstitute;
+
+namespace SimpleCalculatorTest;
+
+public class Test
+{
+    [Fact]
+    public void GetOperator_DivisionEnum_DivisionOperator() {
+        IOperatorProvider operatorProvider = new OperatorProvider();
+
+        IOperator returnedOperator = operatorProvider.GetOperator(OperatorEnum.division);
+
+        Assert.IsType<DivisionOperator>(returnedOperator);
+    }
+
+    [Fact]
+    public void GetOperator_MultiplyEnum_MultiplyOperator() {
+        IOperatorProvider operatorProvider = new OperatorProvider();
+
+        IOperator returnedOperator = operatorProvider.GetOperator(OperatorEnum.multiply);
+
+        Assert.IsType<MultiplyOperator>(returnedOperator);
+    }
+
+    [Fact]
+    public void GetOperator_SumEnum_SumOperator() {
+        IOperatorProvider operatorProvider = new OperatorProvider();
+
+        IOperator returnedOperator = operatorProvider.GetOperator(OperatorEnum.sum);
+
+        Assert.IsType<SumOperator>(returnedOperator);
+    }
+
+
+    [Fact]
+    public void GetOperator_SubEnum_SubOperator() {
+        IOperatorProvider operatorProvider = new OperatorProvider();
+
+        IOperator returnedOperator = operatorProvider.GetOperator(OperatorEnum.sub);
+
+        Assert.IsType<SubOperator>(returnedOperator);
+    }
+
+    [Fact]
+    public void GetOperator_NullEnum_ThrowsNotSupportedException() {
+        IOperatorProvider operatorProvider = new OperatorProvider();
+        IOperator returnedOperator;
```

Comment:
redundant variable: since we expect that the `GetOperator` itself would throw an exception, we do not need to assign it to the `returnedOperator` variable. call it directly instead:
```cs
Action act = () operatorProvider.GetOperator(...)
```